app/models/enterprise.rb

```diff
@@ -408,7 +408,7 @@ def initialize_permalink
   end
 
   def touch_distributors
-    Enterprise.distributing_products(self.supplied_products).
+    Enterprise.distributing_products(self.supplied_products.select(:id)).
```

Comment:
We don't need self here, so we could remove it while we are at it.

Reply:
ok, done